### ISSUE TITLE
Build error with 1.3.8 (doc)

### ISSUE THREAD

**Issue by turbo** [rust-lang/regex#684]
This crate is used somewhere in the dep tree of a package I need (tokei). It seems like the latest update broke the build for my current rust version (it did work a few hours ago). I'll investigate, but this is what's happening:

```text
Compiling rand_xorshift v0.1.1
Compiling phf v0.7.24
Compiling aho-corasick v0.7.10
Compiling libloading v0.5.2
Compiling stringprep v0.1.2
Compiling regex v1.3.8
Compiling quote v1.0.6
error[E0658]: `cfg(doctest)` is experimental and subject to change
 --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-1.3.8/src/lib.rs:625:7
625 #[cfg(doctest)]
^^^^^^^
note: for more information, see https://github.com/rust-lang/rust/issues/62210

error[E0658]: `cfg(doctest)` is experimental and subject to change --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-1.3.8/src/lib.rs:636:7

#[cfg(doctest)]
^^^^^^^
note: for more information, see https://github.com/rust-lang/rust/issues/62210

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0658`.

error: could not compile `regex`.

warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by turbo** [comment]
Caused by this change: https://github.com/rust-lang/regex/commit/f98eea5d9eb97e4fc9efcf31963364eb54f98610

**Comment by BurntSushi** [comment]
This ticket isn't actionable. I see no way to reproduce it. I don't see which version of Rust you're using nor which commands you've run. Could you please provide enough details to reproduce the problem?

**Comment by turbo** [comment]
This was stabilized in `1.43.1`. Apparently my version is too old. I shall update it.

**Comment by BurntSushi** [comment]
`regex` compiles just fine for me on Rust 1.28, which is its MSRV. So it's still not clear to me where the problem is.

**Comment by BurntSushi** [comment]
This was an interesting issue, but it is now fixed! `regex 1.3.8` compiled on Rust 1.28 but did not compile on, e.g., Rust 1.39. That explains why I wasn't able to reproduce this, since I tried building with Rust 1.28 (which CI also does).

`regex 1.3.9` should restore the MSRV of Rust 1.28.

**Comment by turbo** [comment]
Perfect timing, I was just now able to restore access to the Dockerfile that included the rust compiler and indeed it was 1.39 :smile: 

Glad you were able to find it despite the lack of info.
